src/t_stream.c

```diff
@@ -1809,20 +1824,140 @@ size_t streamReplyWithRange(client *c, stream *s, streamID *start, streamID *end
+            
+            if (count && listLength(eligible_pels) >= count) break;
+        }
+        raxStop(&ri);
+
+        /* Process each eligible pending entry, claiming it for the current consumer.
+         * For each entry we:
+         * 1. Fetch the actual message data from the stream
+         * 2. Send the message to the client with metadata (idle time, delivery count)
+         * 3. Transfer ownership from the previous consumer to the current consumer
+         * 4. Update all relevant data structures and propagate the claim operation */
+        listIter li;
+        listNode *ln;
+        listRewind(eligible_pels, &li);
+        while ((ln = listNext(&li))) {
+            pelTimeKey *pelKey = (pelTimeKey*)listNodeValue(ln);
+            unsigned char buf[sizeof(streamID)];
+            streamEncodeID(buf, &pelKey->id);
+
+            void *result;
+            streamNACK *nack = NULL;
+            uint64_t delivery_count = 0;
+            /* Must exist, we got the ID from pel_by_time */
+            serverAssert(raxFind(group->pel,buf,sizeof(buf),&result));
+
+            nack = (streamNACK*)result;
+            delivery_count = nack->delivery_count;
+
+            streamID pel_id;
+            streamIteratorStart(&si,s,&pelKey->id,&pelKey->id,rev);
+            if (streamIteratorGetID(&si,&pel_id,&numfields)) {
+                /* Emit a four elements array: ID, array of field-value pairs,
+                 * idle time and delivery count. */
+                addReplyArrayLen(c,4);
+                addReplyStreamID(c,&pel_id);
+                addReplyArrayLen(c,numfields*2);
+
+                /* Emit the field-value pairs. */
+                while (numfields--) {
+                    unsigned char *key, *value;
+                    int64_t key_len, value_len;
+                    streamIteratorGetField(&si,&key,&value,&key_len,&value_len);
+                    addReplyBulkCBuffer(c,key,key_len);
+                    addReplyBulkCBuffer(c,value,value_len);
+                }
+
+                uint64_t idle = commandTimeSnapshot() - pelKey->delivery_time;
+                addReplyBulkLongLong(c, idle);
+                addReplyBulkLongLong(c, delivery_count);
+
+                /* Remove the NACK from old consumer.*/
+                raxRemove(nack->consumer->pel,buf,sizeof(buf),NULL);
+                /* Remove the NACK from the PEL by time. */
+                raxRemovePelByTime(group->pel_by_time, nack->delivery_time, &pel_id);
+                /* Update the consumer and NACK metadata. */
+                nack->consumer = consumer;
+                nack->delivery_time = commandTimeSnapshot();
+                nack->delivery_count++;
+                /* Add the entry in the new consumer local PEL. */
+                raxInsert(consumer->pel,buf,sizeof(buf),nack,NULL);
+                
+                /* Add updated NACK in PEL by time. */
+                raxInsertPelByTime(group->pel_by_time, nack->delivery_time, &pel_id);
```

Comment:
just suggest, I feel that code segmentation makes it more readable, I always think it's too fragmented now.


```suggestion
```suggestion
                /* Remove the NACK from old consumer and time-based PEL. */
                raxRemove(nack->consumer->pel,buf,sizeof(buf),NULL);
                raxRemovePelByTime(group->pel_by_time, nack->delivery_time, &pel_id);

                /* Transfer NACK to new consumer with updated metadata.  */
                nack->consumer = consumer;
                nack->delivery_time = commandTimeSnapshot();
                nack->delivery_count++;
                raxInsert(consumer->pel,buf,sizeof(buf),nack,NULL);
                raxInsertPelByTime(group->pel_by_time, nack->delivery_time, &pel_id);
```

Reply:
Done.